src/traces/histogram/calc.js

```diff
@@ -176,15 +176,17 @@ module.exports = function calc(gd, trace) {
                 b: 0
             };
 
-            // pts and p0/p1 don't seem to make much sense for cumulative distributions
+            // setup hover and event data fields,
+            // N.B. pts and "hover" positions ph0/ph1 don't seem to make much sense
+            // for cumulative distributions
             if(!cumulativeSpec.enabled) {
                 cdi.pts = inputPoints[i];
                 if(uniqueValsPerBin) {
-                    cdi.p0 = cdi.p1 = (inputPoints[i].length) ? pos0[inputPoints[i][0]] : pos[i];
+                    cdi.ph0 = cdi.ph1 = (inputPoints[i].length) ? pos0[inputPoints[i][0]] : pos[i];
```

Comment:
ping @alexcjohnson I hope you don't mind.

The bar corners p0/p1/s0/s1 are now set during `Bar.crossTraceCalc` (moved from `Bar.plot`) so that we don't have to duplicate logic in `BarPolar.plot`. In turn, we needed to rename histogram p0/p1 to not break histogram hover.